build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/api.rb

```diff
@@ -7,9 +7,15 @@ class << self
       # @return [String, nil]
       def docstring(shape_or_shape_ref, api)
         ref, shape = resolve(shape_or_shape_ref, api)
+        # APIG models, downcase shape name in origin or "__" prefix in origin
         if shape.nil?
-          # APIG models, downcase shape name in origin
           ref, shape = resolve(AwsSdkCodeGenerator::Helper.downcase_first(shape_or_shape_ref), api)
+          if shape.nil?
```

Comment:
This is probably a place where comments are worthwhile, to explain why we nest this twice.